src/Cli/dotnet/CommonOptions.cs

```diff
@@ -100,6 +115,38 @@ public static bool VerbosityIsDetailedOrDiagnostic(this VerbosityOptions verbosi
                 verbosity.Equals(VerbosityOptions.d) ||
                 verbosity.Equals(VerbosityOptions.detailed);
         }
+
+        internal static IEnumerable<string> ResolveArchOptionToRuntimeIdentifier(string arg)
+        {
+            var currentRid = GetCurrentRuntimeId();
+            var os = currentRid.Substring(0, currentRid.LastIndexOf("-"));
+            return new string[] { $"-property:RuntimeIdentifier={os}-{arg}", "-property:SelfContained=true" };
+        }
+
+        internal static IEnumerable<string> ResolveOsOptionToRuntimeIdentifier(string arg)
+        {
+            var currentRid = GetCurrentRuntimeId();
+            var arch = currentRid.Substring(currentRid.LastIndexOf("-") + 1, currentRid.Length - currentRid.LastIndexOf("-") - 1);
+            return new string[] { $"-property:RuntimeIdentifier={arg}-{arch}", "-property:SelfContained=true" };
+        }
+
+        private static string GetCurrentRuntimeId()
```

Comment:
There is a lot of logic here. It seems like this method would be a lot easier to use if you added a tiny bit more logic, split the RID and then returned a tuple or a struct. This line would be longer be needed with each usage: https://github.com/dotnet/sdk/pull/18889/files#diff-99e0f27176e2b38367e3898ec98c6d1a67f2cb629195156f4bc1c57b2ef6670bR129.